### ISSUE TITLE
Extension specifying `extensionPack` with non-web-compatible ext's cannot be installed in vscode.dev even if the wrapper ext is web-compatible

### ISSUE THREAD

**Issue by jonathanyeung** [microsoft/vscode#157805]
I just upgraded my extension to try to get it to work on vscode on the web. I added a new entry point for 'browser' in my package.json.

When viewing in the extensions tab in the browser, my extension now no longer shows the yellow exclamation mark of browser incompatibility and I get an option to 'Install', so it seems like _something_ worked. But when I actually try to install it, I get an error:

![](https://user-images.githubusercontent.com/3203268/183872789-fabdcf87-d9a1-4ac9-82c5-5d39fc8ec2dd.png)

From my local testing + opening up the marketplace .vsix file, the 'package.json' manifest and bundle seem to be configured correctly. It appears that the extension is being gated on the Marketplace side - any ideas or pointers?

The extension in question is latest version (v107.2) of https://marketplace.visualstudio.com/items?itemName=dendron.nightly

vsce version: 2.10.0

Thanks!

---

Version: 1.70.0
Commit: da76f93349a72022ca4670c1b84860304616aaa2
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.5112.81 Safari/537.36 Edg/104.0.1293.47
Embedder: vscode.dev

**Comment by jonathanyeung** [comment]
Hi @sandy081 - any updates on this one? This is blocking our release - thanks!

**Comment by jonathanyeung** [comment]
Ok - I figured out the issue.  We have the following in our `package.json`: 

```json
  "extensionPack": [
    "dendron.dendron-paste-image"
  ],
  ```

The `dendron.dendron-paste-image` package hasn't been upgraded to a web extension yet.

However, IMO to me this is an issue with the marketplace validation - I would expect in this case to get a dialog like the following: https://code.visualstudio.com/docs/editor/vscode-web#_extensions ('Python' has limited functionality in VSCode for the web).  Although none of the extensions inside the extensionPack are web-compatible, the extensionPack wrapper ext _itself is_ web-compatible in our case.

@VS Code team - can you consider a fix for this. 

Thanks!

**Comment by sandy081** [comment]
> However, IMO to me this is an issue with the marketplace validation - I would expect in this case to get a dialog like the following: https://code.visualstudio.com/docs/editor/vscode-web#_extensions ('Python' has limited functionality in VSCode for the web). Although none of the extensions inside the extensionPack are web-compatible, the extensionPack wrapper ext itself is web-compatible in our case.

Unfortunately Extension pack is VS Code concept and not a Marketplace concept. So you have to make sure the extension included is a web extension.

> Another issue I just found is that even if I updated the dendron.dendron-paste-image extension to be browser compatible, the nightly extension pack extension still won't install until I push a dummy update for it.

I do not think it is required. We always try to install the latest version of the extension. So when you install the pack and if the included extension is supported we install it.
